### PR TITLE
Use Constraints for Dependencies during install

### PR DESCRIPTION
**Pull request by trevormorgan**
using pip install -c constraints_file .[dep_catagory]

### CODE REVIEW

File: .github/workflows/pytest.yaml [hunk 1]
```diff
@@ -44,12 +44,12 @@ jobs:
       uses: actions/cache@v3.0.6
       with:
         path: ${{ steps.prepare-cache-pip.outputs.cache-dir }}
-        key: ${{ runner.os }}-${{ matrix.python-version }}-pip-${{ hashFiles('**/requirements.txt') }}-${{ hashFiles('**/setup.cfg') }}
+        key: ${{ runner.os }}-${{ matrix.python-version }}-pip-${{ hashFiles('**/requirements.txt') }}-${{ hashFiles('**/pyproject.toml') }}
         restore-keys: |
           ${{ runner.os }}-${{ matrix.python-version }}-pip-
-    - name: install requirements.txt
+    - name: installtest dependancies
```
Comment (jenshnielsen) [inline .github/workflows/pytest.yaml:50]:
```suggestion
    - name: install broadbean with test dependencies 
```

File: .github/workflows/pytest.yaml [hunk 2]
```diff
@@ -44,12 +44,12 @@ jobs:
...
       run: |
-        pip install -r requirements.txt
+        pip install .[test] -c requirements.txt
     - name: install broadbean
```
Comment (jenshnielsen) [inline .github/workflows/pytest.yaml:53]: This can be removed

File: .github/workflows/docs.yaml [hunk 3]
```diff
@@ -61,15 +61,13 @@ jobs:
       uses: actions/cache@v3.0.6
       with:
         path: ${{ steps.prepare-cache-pip.outputs.cache-dir }}
-        key: ${{ runner.os }}-${{ matrix.python-version }}-pip-${{ hashFiles('**/requirements.txt') }}-${{ hashFiles('**/setup.cfg') }}
+        key: ${{ runner.os }}-${{ matrix.python-version }}-pip-${{ hashFiles('**/requirements.txt') }}-${{ hashFiles('**/pyproject.toml') }}
         restore-keys: |
           ${{ runner.os }}-${{ matrix.python-version }}-pip-
 
-    - name: install requirements.txt
+    - name: install broadbean & docs dependancies
```
Comment (jenshnielsen) [inline .github/workflows/docs.yaml:68]:
```suggestion
    - name: install broadbean & docs dependencies 
```

File: .github/workflows/docs.yaml [hunk 4]
```diff
@@ -61,15 +61,13 @@ jobs:
...
       run: |
-        pip install -r requirements.txt
-    - name: install broadbean
-      run: pip install .
+        pip install -c requirements.txt .[docs,test]
```
Comment (jenshnielsen) [inline .github/workflows/docs.yaml:70]: I think we should document this
```suggestion
        pip install -c requirements.txt .[docs,test]
        # currently we are building api docs of the whole package including tests so we need to be able to import the tests and therefore require test dependencies to build the docs. 
```

File: .github/workflows/pytest.yaml (hunk 2 above)
Comment (jenshnielsen) [reply .github/workflows/pytest.yaml:53]: (the extra `pip install .` command )

**Review by jenshnielsen — APPROVED** [review]

File: .github/workflows/docs.yaml (hunk 4 above)
Comment (trevormorgan) [reply .github/workflows/docs.yaml:70]: good call!

**Pull request merged**